src/gsIO/gsXml.hpp

```diff
@@ -66,24 +67,36 @@ char * makeValue(const gsMatrix<T> & value, gsXmlTree & data,
     return data.allocate_string( oss.str().c_str() );
 }
 
-template<class T>
-void getMatrixFromXml ( gsXmlNode * node, unsigned const & rows,
-                        unsigned const & cols, gsMatrix<T> & result )
-{
-    //gsWarn<<"Reading "<< node->name() <<" matrix of size "<<rows<<"x"<<cols<<"Geometry..\n";
-    std::istringstream str;
-    str.str( node->value() );
-    result.resize(rows,cols);
-
-    for (unsigned i=0; i<rows; ++i) // Read is RowMajor
-        for (unsigned j=0; j<cols; ++j)
-            //if ( !(str >> result(i,j) ) )
-              if (! gsGetValue(str,result(i,j)) )
-            {
-                gsWarn<<"XML Warning: Reading matrix of size "<<rows<<"x"<<cols<<" failed.\n";
-                gsWarn<<"Tag: "<< node->name() <<", Matrix entry: ("<<i<<", "<<j<<").\n";
-                return;
-            }
+template <class T>
+void getMatrixFromXml(gsXmlNode* node, unsigned const& rows,
+                      unsigned const& cols, gsMatrix<T>& result,
+                      const std::string& base_type_flag) {
+    // Make sure that flag is in lower case for comparisons
+    std::string base_type_flag_;
+    base_type_flag_.reserve(base_type_flag.size());
+    std::transform(base_type_flag.cbegin(), base_type_flag.cend(),
+                   std::back_inserter(base_type_flag_),
+                   [](unsigned char c) { return std::tolower(c); });
```

Comment:
Please use "::tolower" instead of "std::tolower" because some windows compiles like it better.